### PR TITLE
Support content attachments API

### PR DESCRIPTION
**Pull request by dreamer-89**

### CODE REVIEW

**Comment by gmlewis** [comment]
Hi @dreamer-89 - the first Travis CI failure is due to `github-accessors.go` not being updated by `go generate ./...` - see CONTRIBUTING.md for details on how to run this.

The problem with the unit test appears to be that on line 192 you are missing a leading slash:
`"content_references/11/attachments"` should be `"/content_references/11/attachments"`.

File: github/apps.go [hunk 1]
```diff
@@ -183,6 +189,29 @@ func (s *AppsService) CreateInstallationToken(ctx context.Context, id int64) (*I
 	return t, resp, nil
 }
 
+// Create a new attachment on user comment containing a url
+//
+// Github API docs: https://developer.github.com/v3/apps/#create-a-content-attachment
```
Comment (gmlewis) [inline github/apps.go]: s/Github/GitHub/

File: github/apps.go [hunk 2]
```diff
@@ -183,6 +189,29 @@ func (s *AppsService) CreateInstallationToken(ctx context.Context, id int64) (*I
 	return t, resp, nil
 }
 
+// Create a new attachment on user comment containing a url
```
Comment (gmlewis) [inline github/apps.go]: Please end sentence with a period for the godocs.

File: github/apps.go [hunk 3]
```diff
@@ -183,6 +189,29 @@ func (s *AppsService) CreateInstallationToken(ctx context.Context, id int64) (*I
...
+//
+// Github API docs: https://developer.github.com/v3/apps/#create-a-content-attachment
+func (s *AppsService) CreateAttachment(ctx context.Context, contentReferenceID int64, title, body string) (*Attachment, *Response, error) {
+	u := fmt.Sprintf("content_references/%v/attachments", contentReferenceID)
+	fmt.Println(u)
```
Comment (gmlewis) [inline github/apps.go]: Please delete this line.

File: github/apps.go (hunk 1 above)
Comment (dreamer-89) [reply github/apps.go]: Hi @gmlewis, can you please elaborate the required change ?

Comment (gmlewis) [reply github/apps.go]: Yes, please change "Github" to "GitHub".  Thanks.

Comment (dreamer-89) [reply github/apps.go]: My bad

**Comment by dreamer-89** [comment]
@gmlewis : Changes done as suggested. Please review.

File: github/apps.go [hunk 4]
```diff
@@ -62,6 +62,12 @@ type Installation struct {
 	UpdatedAt           *Timestamp               `json:"updated_at,omitempty"`
 }
 
+type Attachment struct {
```
Comment (gmlewis) [inline github/apps.go]: Please document this new struct similarly to how the other structs in this repo are documented.

File: github/apps_test.go [hunk 5]
```diff
@@ -185,6 +185,27 @@ func TestAppsService_CreateInstallationToken(t *testing.T) {
...
+
+	got, _, err := client.Apps.CreateAttachment(context.Background(), 11, "title1", "body1")
+	if err != nil {
+		t.Errorf("CreateAttachment returned error: %v", err)
+	}
+	want := &Attachment{ID: Int64(1), Title: String("title1"), Body: String("body1")}
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("CreateAttachment = %+v, want %+v", got, want)
+	}
+}
```
Comment (kshitij10496) [inline github/apps_test.go]: suggestion: A newline separating the two unit tests would be visually aesthetic.

**Comment by nadavkaner** [comment]
Can I continue with this PR and do the remaining changes?

**Comment by gmlewis** [comment]
Thank you for the offer, @nadavkaner.

I'm trying to determine the best way to proceed as it appears that this PR has been abandoned. Maybe it is best to merge this one and have you follow up with the requested changes in a new PR.

Do you have any recommendations, @willnorris?

**Comment by nadavkaner** [comment]
I suggesting to do as you said, merging this PR to my own branch. This is the best option i think

**Comment by nadavkaner** [comment]
I tried to merge this branch to my own branch, but because cla restrictions because it contains another person commits it need to get approve from him.
So I just took his changes and created commit with his changes + the review changes:
[#1139](https://github.com/google/go-github/pull/1139)

**Comment by gmlewis** [comment]
Closed by #1139.

**Pull request closed without merging**